File: apps/web/src/types/risk.ts
export type RiskTargetType = "SOURCE_VIDEO" | "RENDER_OUTPUT" | "PUBLISH_DRAFT";
export type RiskSeverity = "LOW" | "MEDIUM" | "HIGH" | "CRITICAL" | "BLOCKING";
export type RiskFlagStatus = "OPEN" | "ACKNOWLEDGED" | "RESOLVED" | "WAIVED" | "REJECTED";
export type OperatorRiskDecisionType = "CONTINUE" | "NEEDS_FIX" | "REJECT" | "ACCEPT_WITH_WARNING";

export type RiskFlag = {
  id: string;
  source_video_id: string;
  target_type: RiskTargetType;
  target_id: string | null;
  flag_type: string;
  severity: RiskSeverity;
  status: RiskFlagStatus;
  title: string | null;
  description: string | null;
  evidence_summary: string | null;
  scan_source: string | null;
  detected_at: string | null;
  resolution_note: string | null;
};

export type RiskGateSummary = {
  can_continue: boolean;
  requires_operator_decision: boolean;
  blocking_reasons: string[];
  highest_severity: RiskSeverity | null;
  open_counts_by_severity: Record<string, number>;
  accepted_with_warning: boolean;
};

export type OperatorRiskDecision = {
  id: string;
  target_type: RiskTargetType;
  target_id: string;
  decision_type: OperatorRiskDecisionType;
  note: string | null;
  decided_by: string | null;
  decided_at: string;
};

export type RiskSummary = {
  target_type: RiskTargetType;
  target_id: string;
  flags: RiskFlag[];
  gate: RiskGateSummary;
  latest_decision: OperatorRiskDecision | null;
};

File: apps/api/src/schemas/risk.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from src.enums import OperatorRiskDecisionType, RiskFlagStatus, RiskFlagType, RiskSeverity, RiskTargetType


class RiskScanRequest(BaseModel):
    target_type: RiskTargetType
    target_id: UUID
    scope: str | None = None


class RiskFlagActionRequest(BaseModel):
    note: str | None = None


class RiskDecisionRequest(BaseModel):
    target_type: RiskTargetType
    target_id: UUID
    decision_type: OperatorRiskDecisionType
    note: str | None = None
    decided_by: str | None = "local_operator"


class RiskGateSummaryResponse(BaseModel):
    can_continue: bool
    requires_operator_decision: bool
    blocking_reasons: list[str]
    highest_severity: RiskSeverity | None
    open_counts_by_severity: dict[str, int]
    accepted_with_warning: bool = False


class RiskFlagResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    workspace_id: UUID
    source_video_id: UUID
    target_type: RiskTargetType
    target_id: UUID | None
    scan_run_id: UUID | None
    flag_type: RiskFlagType
    severity: RiskSeverity
    status: RiskFlagStatus
    title: str | None
    description: str | None
    reason: str | None
    evidence_summary: str | None
    scan_source: str | None
    detected_at: datetime | None
    resolved_at: datetime | None
    resolution_note: str | None
    evidence_json: dict | None
    metadata_json: dict | None
    created_at: datetime
    updated_at: datetime


class RiskFlagListResponse(BaseModel):
    flags: list[RiskFlagResponse]


class OperatorRiskDecisionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    workspace_id: UUID
    source_video_id: UUID
    target_type: RiskTargetType
    target_id: UUID
    decision_type: OperatorRiskDecisionType
    note: str | None
    decided_by: str | None
    decided_at: datetime
    gate_summary_json: dict | None
    metadata_json: dict | None
    created_at: datetime
    updated_at: datetime


class RiskSummaryResponse(BaseModel):
    target_type: RiskTargetType
    target_id: UUID
    flags: list[RiskFlagResponse]
    gate: RiskGateSummaryResponse
    latest_decision: OperatorRiskDecisionResponse | None = None


class RiskScanResponse(RiskSummaryResponse):
    scan_run_id: UUID
